okhttp-clients/src/test/java/com/palantir/remoting3/okhttp/OkHttpClientsTest.java

```diff
@@ -442,13 +446,70 @@ public void handlesInterruptedThreads() throws Exception {
         thread.join();
     }
 
+    @Test
+    public void timeouts_set_to_all_zero_should_be_treated_as_infinity() throws Exception {
+        server.enqueue(new MockResponse()
+                .setBodyDelay(Duration.ofSeconds(11).toMillis(), TimeUnit.MILLISECONDS)
+                .setBody("Hello, world!"));
+
+        OkHttpClient client = OkHttpClients.withStableUris(
+                ClientConfiguration.builder()
+                        .from(createTestConfig(url))
+                        .connectTimeout(Duration.ZERO)
+                        .readTimeout(Duration.ZERO)
+                        .writeTimeout(Duration.ZERO) // want to allow unlimited time for uploads
```

Comment:
if the writeTimeout is the timeout for the client writing to the server, then we aren't testing that in this case, since it's a sync-request

Reply:
True, this test just verifies that slow downloads work.  In practise, since the timeout is at a high enough level (it can't distinguish between uploads vs downloads) I'm confident this will also be fine for uploads.